src/Modal/ModalHeader/index.js

```diff
@@ -157,13 +157,21 @@ class ModalHeader extends React.PureComponent<Props> {
   };
 
   render() {
-    const { title, illustration, description, children, suppressed, dataTest } = this.props;
+    const {
+      title,
+      illustration,
+      description,
+      children,
+      suppressed,
+      dataTest,
+      modalID,
+    } = this.props;
     const hasHeader = title || description;
     return (
       <StyledModalHeader illustration={!!illustration} suppressed={suppressed} data-test={dataTest}>
         {illustration}
         {hasHeader && (
-          <ModalTitle illustration={!!illustration}>
+          <ModalTitle illustration={!!illustration} id={modalID}>
```

Comment:
I think `id` should be in something that is always rendered like `ModalWrapper` not only in `ModalHeader`